lib/action_client/applications/submission.rb

```diff
@@ -0,0 +1,22 @@
+module ActionClient
+  module Applications
+    class Submission
+      def initialize(app, after_action_callbacks)
+        @app = app
+        @after_action_callbacks = Array(after_action_callbacks).compact
+      end
+
+      def call(env)
+        response = @app.call(env)
+
+        @after_action_callbacks.reduce(response) do |(status, headers, body), callback|
+          if callback.arity == 1
+            [status, headers, callback.call(body)]
+          else
+            callback.call(status, headers, body)
```

Comment:
Is this too magical and counter intuitive? Would a separate hook named `transform_body` (or something like that) be preferable to making assumptions of intent on behalf of the caller?